merlin/study/batch.py

```diff
@@ -143,60 +147,70 @@ def batch_worker_launch(spec, com, nodes=None, batch=None):
     if nodes is None or nodes == "all":
         nodes = get_node_count(default=1)
 
-    bank = get_yaml_var(batch, "bank", "")
-    queue = get_yaml_var(batch, "queue", "")
-    shell = get_yaml_var(batch, "shell", "bash")
-    walltime = get_yaml_var(batch, "walltime", "")
+    shell: Union[str, Dict] = get_yaml_var(batch, "shell", "bash")
 
-    launch_pre = get_yaml_var(batch, "launch_pre", "")
-    launch_args = get_yaml_var(batch, "launch_args", "")
-    worker_launch = get_yaml_var(batch, "worker_launch", "")
+    launch_pre: Union[str, Dict] = get_yaml_var(batch, "launch_pre", "")
+    launch_args: Union[str, Dict] = get_yaml_var(batch, "launch_args", "")
+    launches: str = get_yaml_var(batch, "worker_launch", "")
 
-    if btype == "flux":
-        launcher = get_batch_type()
-    else:
-        launcher = get_batch_type()
-
-    launchs = worker_launch
-    if not launchs:
-        if btype == "slurm" or launcher == "slurm":
-            launchs = f"srun -N {nodes} -n {nodes}"
-            if bank:
-                launchs += f" -A {bank}"
-            if queue:
-                launchs += f" -p {queue}"
-            if walltime:
-                launchs += f" -t {walltime}"
-        if launcher == "lsf":
-            # The jsrun utility does not have a time argument
-            launchs = f"jsrun -a 1 -c ALL_CPUS -g ALL_GPUS --bind=none -n {nodes}"
-
-    launchs += f" {launch_args}"
+    if not launches:
+        launches = construct_launches(batch, btype, nodes)
+
+    launches += f" {launch_args}"
```

Comment:
change variable name throughout this function

Reply:
Patched.